src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,1346 @@
+            logging.debug("Target position reached.")
+            self._operationalReached.set()
+        else:
+            self._operationalReached.clear()
+
+    def _updateGasOn(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the GIS gas flow state from the Orsay server and saves it in the gasOn VA
+        """
+        if parameter is None:
+            parameter = self._reservoirPar
+        if parameter is not self._reservoirPar:
+            raise ValueError("Incorrect parameter passed to _updateGasOn. Parameter should be "
+                             "datamodel.HybridGIS.ReservoirState. Parameter passed is %s." % parameter.Name)
+        if not attributeName == "Actual":
+            return
+        logging.debug("Gas flow is now %s." % self._reservoirPar.Actual)
+        new_value = self._reservoirPar.Actual.lower() == "open"
+        self.gasOn._value = new_value  # to not call the setter
+        self.gasOn.notify(new_value)
+
+    def _setGasOn(self, goal):
+        """
+        goal (bool): the goal state of the gas flow: (True: "open", False: "closed")
+        return (bool): the new state the gas flow: (True: "open", False: "closed")
+
+        Opens the GIS reservoir if argument goal is True. Closes it otherwise.
+        Also closes the reservoir if the position of the GIS is not operational.
+        """
+        if not self.position.value["operational"]:
+            logging.warning("Gas flow was attempted to be changed while not in working position.")
+            goal = False
+        if goal:
+            logging.debug("Starting gas flow.")
+            self._reservoirPar.Target = "OPEN"
+        else:
+            logging.debug("Stopping gas flow.")
+            self._reservoirPar.Target = "CLOSED"
+        return self._reservoirPar.Target.lower() == "open"
+
+    def _setOperational(self, goal):
+        """
+        goal (bool): the goal state of the GIS position: (True: "work", False: "park")
+        return (bool): the state the GIS position is in after finishing moving: (True: "work", False: "park")
+
+        Moves the GIS to working position if argument goal is True. Moves it to parking position otherwise.
+        """
+        if self.gasOn.value:
+            logging.warning("Gas flow was still on when attempting to move the GIS. Turning off gas flow.")
+            self.gasOn._set_value(False)  # turn off the gas flow if it wasn't already
+        if goal:
+            logging.debug("Moving GIS to operational position.")
+            self._positionPar.Target = "WORK"
+        else:
+            logging.debug("Moving GIS to parking position.")
+            self._positionPar.Target = "PARK"
+        self._operationalReached.wait()
+        return self._positionPar.Actual.lower() == "work"
```

Comment:
Normally a move command/Future doesn't return any value. However, you should update .position.